akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -371,6 +371,110 @@ final case class Scan[In, Out](zero: Out, f: (Out, In) ⇒ Out) extends GraphSta
 /**
  * INTERNAL API
  */
+final case class ScanAsync[In, Out](zero: Out, f: (Out, In) ⇒ Future[Out]) extends GraphStage[FlowShape[In, Out]] {
+
+  import akka.dispatch.ExecutionContexts
+
+  val in = Inlet[In]("ScanAsync.in")
+  val out = Outlet[Out]("ScanAsync.out")
+  override val shape: FlowShape[In, Out] = FlowShape[In, Out](in, out)
+
+  override val initialAttributes: Attributes = Attributes.name("scanAsync")
+
+  override val toString: String = "ScanAsync"
+
+  override def createLogic(inheritedAttributes: Attributes): GraphStageLogic =
+    new GraphStageLogic(shape) with InHandler with OutHandler {
+      self ⇒
+
+      private var aggregator: Out = zero
```

Comment:
I find this name peculiar, how is the aggregated value an aggregator? What about `current`?